services/web/server/tests/unit/with_dbs/02/test_project_db.py

```diff
@@ -504,6 +505,215 @@ def test_find_changed_dict_keys(
+                    }
+                },
+                "progress": 100,
+                "runHash": None,
+            },
+            (True, {"outFile"}),
+            id="file-picker outputs changed (file was removed)",
+        ),
+        pytest.param(
+            {
+                "key": "simcore/services/frontend/file-picker",
+                "version": "1.0.0",
+                "label": "test_local.log",
+                "inputs": {},
+                "inputsUnits": {},
+                "inputNodes": [],
+                "parent": None,
+                "thumbnail": "",
+                "outputs": {
+                    "outFile": {
+                        "store": "0",
+                        "path": "6b96a29a-d73c-11ec-943f-02420a000008/2b5cc601-95dd-4c67-b6b9-c4cf3adcd4d1/test_local.log",
+                        "label": "test_local.log",
+                        "dataset": "6b96a29a-d73c-11ec-943f-02420a000008",
+                    }
+                },
+                "progress": 100,
+            },
+            {
+                "key": "simcore/services/frontend/file-picker",
+                "version": "1.0.0",
+                "label": "test_local.log",
+                "inputs": {},
+                "inputsUnits": {},
+                "inputNodes": [],
+                "parent": None,
+                "thumbnail": "",
+                "outputs": {
+                    "outFile": {
+                        "store": "0",
+                        "path": "6b96a29a-d73c-11ec-943f-02420a000008/2b5cc601-95dd-4c67-b6b9-c4cf3adcd4d1/test_local.log",
+                        "label": "test_local.log",
+                        "dataset": "6b96a29a-d73c-11ec-943f-02420a000008",
+                    }
+                },
+                "progress": 100,
+                "runHash": None,
+            },
+            (False, set()),
+            id="file-picker outputs did not change",
+        ),
+        pytest.param(
+            {"key": "simcore/services/frontend/file-picker", "outputs": None},
+            {"key": "simcore/services/frontend/diffrenet", "outputs": None},
+            (False, set()),
+            id="different keys do not trigger",
+        ),
+        pytest.param(
+            {"key": "simcore/services/frontend/file-picker", "outputs": {}},
+            {"key": "simcore/services/frontend/file-picker", "outputs": None},
+            (True, set()),
+            id="no and not existing outputs trigger",
+        ),
+        pytest.param(
+            {},
+            {},
+            (False, set()),
+            id="all keys missing do not trigger",
+        ),
+        pytest.param(
+            {"a": "key"},
+            {"another": "key"},
+            (False, set()),
+            id="different keys but missing key and outputs do not trigger",
+        ),
+        pytest.param(
+            {"key": "simcore/services/frontend/file-picker"},
```

Comment:
if I get that correctly now this function will be called from comp_tasks_listening_task and also from PATCH /projects right?
is there any test for the case of computational services, and/or other dynamic services? or is this purely and only for file-picker? in which case please change the name of your functions to reflect that (if not this would be confusing).

Reply:
Yes, this affects `comp_tasks_listening_task` and `PATCH /projects` (aka, when the fronted saves the project)

The only affects services are frontend services. And also, for now only the file-picker is involved. I've renamed the functions to highlight this.